pkg/registry/client_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"k8s.io/helm/pkg/chart"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/docker/distribution/configuration"
+	"github.com/docker/distribution/registry"
+	_ "github.com/docker/distribution/registry/storage/driver/inmemory"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type RegistryClientTestSuite struct {
+	suite.Suite
+	Out                io.Writer
+	DockerRegistryHost string
+	RegistryClient     *Client
+}
+
+func (suite *RegistryClientTestSuite) SetupSuite() {
+	os.RemoveAll("helm-registry-test")
```

Comment:
Would this be better in an `AfterTest()`? https://godoc.org/github.com/stretchr/testify/suite#AfterTest

That way we can get rid of the .gitignore file as well.

Reply:
done